src/surface.rs

```diff
@@ -121,7 +121,7 @@ where
         }
     });
     surface.as_ref().user_data().set_threadsafe(|| Mutex::new(SurfaceUserData::new()));
-    surface.detach()
+    (*surface).clone()
```

Comment:
In theory such thing could be implemented in wayland-client to convert `Main<I>` to `Attached<I>`, since `Main<I>` just encapsulates `Attached<I>`.